### DISCUSSION TITLE
[✨] qwik-city: useLocation should provide makePath

### DISCUSSION THREAD

**Discussion by wmertens** [QwikDev/qwik-evolution#146] [Proposals For Qwik]
### Is your feature request related to a problem?

When trying to redirect within the application, it's hard to do it properly.

### Describe the solution you'd like

`useLocation` should return `makePath` function that takes `path` and/or `params` and/or `query`, merging those to return the correct absolute path for the application.

```tsx
const {makePath, params} = useLocation()
return <div>
  <Link href={makePath({path:".."})}>Up</Link>
  <Link href={makePath({params: {id: Number(params.id)-1}})}>Previous</Link>
</div>
```

### Describe alternatives you've considered

Adhoc path generation. People will often forget query strings and the base path

**Comment by manucorporat** [comment]
I don't think this belong in useLocation()

**Comment by wmertens** [comment]
@manucorporat is it because it would create overhead? How about a useMakePath() hook that calls useLocation?

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by wmertens** [comment]
I think I don't like this idea any more. Just writing the paths or creating a helper is simple enough.
